CMake/HighFiveConfig.cmake.in

```diff
@@ -35,35 +35,32 @@ endif()
 
 # If the user sets this flag, all dependencies are preserved.
 # Useful in central deployments where dependencies are not prepared later
-option(HIGHFIVE_USE_INSTALL_DEPS "Use original Highfive dependencies" @HIGHFIVE_USE_INSTALL_DEPS@)
+set(HIGHFIVE_USE_INSTALL_DEPS @HIGHFIVE_USE_INSTALL_DEPS@ CACHE BOOL "Use original Highfive dependencies")
 if(HIGHFIVE_USE_INSTALL_DEPS)
   # If enabled in the deploy config, request c++14
   if(@HIGHFIVE_USE_XTENSOR@ AND NOT CMAKE_VERSION VERSION_LESS 3.8)
     set_property(TARGET HighFive APPEND PROPERTY INTERFACE_COMPILE_FEATURES cxx_std_14)
   endif()
-  message("HIGHFIVE @PROJECT_VERSION@: Using original dependencies (HIGHFIVE_USE_INSTALL_DEPS=YES)")
+  message(STATUS "HIGHFIVE @PROJECT_VERSION@: Using original dependencies (HIGHFIVE_USE_INSTALL_DEPS=YES)")
   copy_interface_properties(HighFive HighFive_HighFive)
   return()
 endif()
 
 # When not using the pre-built dependencies, give user options
-option(HIGHFIVE_USE_BOOST "Enable Boost Support" ON)
-option(HIGHFIVE_USE_EIGEN "Enable Eigen testing" OFF)
-option(HIGHFIVE_USE_XTENSOR "Enable xtensor testing" OFF)
-
-# Options when used from external projects. Keep defaults
-if(NOT DEFINED HIGHFIVE_PARALLEL_HDF5)
-  option(HIGHFIVE_PARALLEL_HDF5 "Enable Parallel HDF5 support" @HIGHFIVE_PARALLEL_HDF5@)
-endif()
-if(NOT DEFINED HIGHFIVE_USE_BOOST)
-  option(HIGHFIVE_USE_BOOST "Enable Boost Support" @HIGHFIVE_USE_BOOST@)
+if(DEFINED HIGHFIVE_USE_BOOST)
+  set(HIGHFIVE_USE_BOOST ${HIGHFIVE_USE_BOOST} CACHE BOOL "Enable Boost Support")
+else()
+  set(HIGHFIVE_USE_BOOST @HIGHFIVE_USE_BOOST@ CACHE BOOL "Enable Boost Support")
 endif()
+set(HIGHFIVE_USE_EIGEN "${HIGHFIVE_USE_EIGEN}" CACHE BOOL "Enable Eigen testing")
+set(HIGHFIVE_USE_XTENSOR "${HIGHFIVE_USE_XTENSOR}" CACHE BOOL "Enable xtensor testing")
```

Comment:
```suggestion
set(HIGHFIVE_USE_EIGEN @HIGHFIVE_USE_EIGEN@ CACHE BOOL "Enable Eigen testing")
set(HIGHFIVE_USE_XTENSOR @HIGHFIVE_USE_XTENSOR@ CACHE BOOL "Enable xtensor testing")
```

Reply:
This PR  ` attempt to respect the previous non-cached variables`. The suggestion you make breaks that. Could you explain your rationale?

Reply:
This is how I understand the docs

> Sets the given cache <variable> (cache entry). Since cache entries are meant to provide user-settable values this does not overwrite existing cache entries by default. Use the FORCE option to overwrite existing entries.

But maybe I'm wrongly interpreting "existing cache entries" ?

Reply:
When user does `-DXXX=YYY` then it's a cache variable and then indeed we didn't have to do the `if`. But, as you try in your case, users may set normal variables from another project and expect HighFive to respect
```
set(HIGHFIVE_USE_BOOST False)
find_package(HighFive)
```
HIGHFIVE_USE_BOOST is in this case a normal variable and therefore the `set CACHE` will create a new CACHED variable and we need the default.

Reply:
If that is the case than the if-statement should be around *all* options no?

Reply:
I tested, and it seems to work fine for *xtensor* and *Eigen*, so I guess I partly miss-understand how things worked. On the other hand (not something I use though) `HIGHFIVE_PARALLEL_HDF5` is (now?) hard-coded as install-time. @ferdonline